src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Registry.java

```diff
@@ -50,7 +51,10 @@ ModuleFile getModuleFile(
    * by {@code key} should be materialized as a repo.
    */
   RepoSpec getRepoSpec(
-      ModuleKey key, ExtendedEventHandler eventHandler, DownloadManager downloadManager)
+      ModuleKey key,
+      ImmutableMap<String, Optional<Checksum>> moduleFileRegistryHashes,
```

Comment:
super-nit: `moduleFileHashes`? The current name suggests to me that the hashes are of registries, somehow.